File: soccer-league/soccer-league/Data/Mappers/PlayerMapper.swift
import Foundation

protocol PlayerMappable {
    func toDomain(_ dto: PlayerDTO) -> Player
}

struct PlayerMapper {
    static func toDomain(_ dto: PlayerDTO) -> Player {
        Player(
            id: dto.id,
            nickname: dto.name ?? "Unknown",
            firstName: dto.firstName,
            lastName: dto.lastName,
            imageURL: dto.imageUrl.flatMap { URL(string: $0)?.pandaScoreThumbnail }
        )
    }
}

extension PlayerMapper: PlayerMappable {
    func toDomain(_ dto: PlayerDTO) -> Player { PlayerMapper.toDomain(dto) }
}

File: csgo-leagues-android/app/src/main/java/com/voidx/csgo/data/mapper/PlayerMapper.kt
package com.voidx.csgo.data.mapper

import com.voidx.csgo.data.dto.PlayerDTO
import com.voidx.csgo.domain.entity.Player

object PlayerMapper {
    fun toDomain(dto: PlayerDTO): Player =
        Player(
            id = dto.id,
            nickname = dto.name?.takeIf(String::isNotBlank),
            fullName = buildList {
                dto.firstName?.takeIf(String::isNotBlank)?.let(::add)
                dto.lastName?.takeIf(String::isNotBlank)?.let(::add)
            }.joinToString(" ").ifBlank { null },
            imageUrl = dto.imageUrl.toPandaScoreThumbnailUrl(),
        )
}
